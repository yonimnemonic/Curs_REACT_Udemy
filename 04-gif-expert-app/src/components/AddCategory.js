import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');//string vacio

    const handleInputChange = (e)=>{
        setInputValue(e.target.value);

        console.log('handleInputChange llamado...')
    }
    const handleSubmit = (e)=>{
        e.preventDefault(); //previene el submit por defecto del formulario
        console.log('Handle Sumit', inputValue)

        if( inputValue.trim().length > 2){//Verificas que si el imput value es mayor a 2 letras entonces modificas el hook
            setCategories(cats => [inputValue, ...cats ]);
            setInputValue('')
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input
                type="text"
                value={ inputValue}
                onChange={ handleInputChange}
            />  
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
