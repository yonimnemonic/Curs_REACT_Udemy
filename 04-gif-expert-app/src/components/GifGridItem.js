import React from 'react';
import PropTypes from 'prop-types'

export const GifGridItem = ({title, url}) => { //crea la caja con la imagen y titulo

    return (
        <div className='card animate__animated animate__fadeIN'>
            <img src={ url } alt={ title }></img>
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}


/*
1-AÃ±adir proptypes- url- title
2- enviar el title y el url a la hora de utiliizar shallow()
3-actualizar su snapshot con los cambios.
*/