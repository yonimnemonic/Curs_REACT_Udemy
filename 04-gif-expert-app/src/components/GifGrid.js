import React from 'react';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const { data: images,  loading} = useFetchGifs( category );

    // useEffect( ()=>{
    //     GetGifs(category)
    //     .then( setImages)
    // }, [ category] ) //dipara el código solo cuando se renderiza el componente por 1ª vez.

    return (
        <>
        <h3>{category}</h3>
        {loading && <p>Loading</p>}
        <div className='card-grid'>
            {
             images.map( ( img ) =>(
                 <GifGridItem 
                 img={img.id}
                 { ...img }
                 />
                 ))
            }
        </div>
        
        </>
    )
}
