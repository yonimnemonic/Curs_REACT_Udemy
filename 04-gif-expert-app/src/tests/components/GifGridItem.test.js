import { shallow } from "enzyme";
import react from "react";
import { GifGridItem } from "../../components/GifGridItem";


describe('Pruebas en  <GifGridItem />', () => {

  const title = 'Papito';
  const url = 'https://localhost/ed.jpg'
  const wrapper = shallow( <GifGridItem title={title} url={url} />);

  test('Se debe renderizar correctamente', () => {
    
     expect(wrapper).toMatchSnapshot();
  });

  test('Debe tener un pÃ¡rrafo con el title', () => {
    
    const p = wrapper.find('p');
    expect( p.text().trim()).toBe( title );
  });

  test('Debe tener la imagen igual al URL de los props', () => {
     const img = wrapper.find('img');
     console.log( img.prop('src') )
     expect(img.prop('src')).toBe(url);
     expect(img.prop('alt')).toBe(title);

  });
  test('Debe tener animate_fadeIn', () => {
    const div= wrapper.find('div');
    const className = div.prop('className');
    
    expect(className.includes('animate_fadeIn')).toBe(false);
    
  });
  
  
  
});

