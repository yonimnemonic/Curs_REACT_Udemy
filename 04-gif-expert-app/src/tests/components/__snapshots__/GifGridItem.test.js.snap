// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <GifGridItem /> Se debe renderizar correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIN"
>
  <img
    alt="Papito"
    src="https://localhost/ed.jpg"
  />
  <p>
    Papito
  </p>
</div>
`;
